third_party/chromium/media/formats/mp4/box_definitions.cc

```diff
@@ -1438,6 +1438,120 @@ bool OpusSpecificBox::Parse(BoxReader* reader) {
+  bool obu_redundant_copy = (header >> 2) & 1;
+  bool obu_trimming_status_flag = (header >> 1) & 1;
+  bool obu_extension_flag = header & 1;
+
+  redundant_copy |= obu_redundant_copy;
+
+  uint32_t obu_size;
+  int num_leb_128_bytes;
+  RCHECK(reader->ReadLeb128(&config_obus[buffer_pos], &obu_size, &num_leb_128_bytes));
+  buffer_pos += num_leb_128_bytes;
+
+  size_t obu_stop_pos = reader->pos() + obu_size;
+
+  if (obu_trimming_status_flag) {
+    uint32_t leb_128_value;
+    // num_samples_to_trim_at_end.
+    RCHECK(reader->ReadLeb128(&config_obus[buffer_pos], &leb_128_value, &num_leb_128_bytes));
+    buffer_pos += num_leb_128_bytes;
+    // num_samples_to_trim_at_start.
+    RCHECK(reader->ReadLeb128(&config_obus[buffer_pos], &leb_128_value, &num_leb_128_bytes));
+    buffer_pos += num_leb_128_bytes;
+  }
+
+  if (obu_extension_flag) {
+    uint32_t extension_header_size;
+    RCHECK(reader->ReadLeb128(&config_obus[buffer_pos], &extension_header_size, &num_leb_128_bytes));
+    buffer_pos += num_leb_128_bytes;
+    for (int i = 0; i < extension_header_size; ++i) {
+      RCHECK(reader->Read1(&config_obus[buffer_pos]));
+      buffer_pos++;
+    }
+  }
+
+  switch(static_cast<int>(obu_type)) {
+    case kIAMFObuTypeCodecConfig:
+      MEDIA_LOG(INFO, reader->media_log()) << "OBU type Codec Config.";
+      break;
+    case kIAMFObuTypeAudioElement: {
+      MEDIA_LOG(INFO, reader->media_log()) << "OBU type Audio Element.";
+      break;
+    }
+    case kIAMFObuTypeMixPresentation: {
+      MEDIA_LOG(INFO, reader->media_log()) << "OBU type Mix Presentation.";
+      break;
+    }
+    case kIAMFObuTypeSequenceHeader: {
+      MEDIA_LOG(INFO, reader->media_log()) << "OBU type Sequence Header.";
+      // RCHECK(!read_first_sequence_header);
+      // read_first_sequence_header = true;
+
+      uint32_t ia_code;
+      RCHECK(reader->Read4(&ia_code));
+      RCHECK(ia_code == FOURCC_IAMF);
+      memcpy(&config_obus[buffer_pos], &ia_code, sizeof(ia_code));
+      buffer_pos += sizeof(ia_code);
+
+      // Read primary and additional profiles.
+      RCHECK(reader->Read1(&primary_profile));
+      RCHECK(reader->Read1(&additional_profile));
+      memcpy(&config_obus[buffer_pos], &primary_profile, sizeof(primary_profile));
+      buffer_pos += sizeof(primary_profile);
+      memcpy(&config_obus[buffer_pos], &additional_profile, sizeof(additional_profile));
+      buffer_pos += sizeof(additional_profile);
+
+      // MEDIA_LOG(INFO, reader->media_log()) << "Primary profile "
+      //                                      << static_cast<int>(primary_profile)
+      //                                      << ", additional profile "
+      //                                      << static_cast<int>(additional_profile)
+      //                                      << ", IA code " << std::hex << ia_code;
+      break;
+    }
+    // We don't handle Data OBUs here.
+    default: {
+      RCHECK(static_cast<int>(obu_type) < kIAMFObuTypeMax);
```

Comment:
This code fails when |obu_type| is |kIAMFObuTypeSequenceHeader|.

Also, since obu types are not continuous, avoid checking using it's max value in general.  Consider listing them separately without a default statement, so compiler can generate warnings whenever there is a mismatch.